### ISSUE TITLE
Schedule Charging Not Working

### ISSUE THREAD

**Issue by MooneyDW** [gndean/home-assistant-hypervolt-charger#38]
Integration works well apart from Schedule Charging. If I try to set up a schedule I get the error "Failed to call service/select_option. 'NoneType' object is not iterable".

**Comment by gndean** [comment]
I'm not sure what this error is related to.
Can you describe exactly what you are doing when this error happens? Are you:
*  setting the activation mode to Scheduled?, or
* Changing a schedule start or end time, or
* Pressing the "Apply Schedule" button?, or
* something else

Is this from the HA user interface? Or via automation?

**Comment by MooneyDW** [comment]
Hi and thanks for making contact.

I was attempting to set up scheduling from the HA interface but I don’t think I had installed the extension properly. I reinstalled it and now everything seems to be working as it should be. Thank you for the extension.

Could I ask a question? What I would like to do with the Hypervolt charger is to be able to assign a charging mode to each individual schedule. Eg: Boost mode for the time I get a low “ev” electricity rate and “Super Eco” mode to only use surplus solar power during the middle of the day.

I have spoken to Hypervolt and they say that this is not possible however I imagine it could be done with automation. Is this possible?

Regards and thanks for your work,

David
